subsys/net/l2/ppp/ppp_l2.c

```diff
@@ -80,6 +80,13 @@ static enum net_verdict process_ppp_msg(struct net_if *iface,
 		return NET_CONTINUE;
 	}
 
+	if ((IS_ENABLED(CONFIG_NET_IPV4) && (protocol >> 8) == PPP_IP) ||
+	    (IS_ENABLED(CONFIG_NET_IPV6) && (protocol >> 8) == PPP_IPV6)) {
+		(void)net_buf_pull_u8(pkt->buffer);
+		net_pkt_cursor_init(pkt);
+		return NET_CONTINUE;
+	}
```

Comment:
Please add comment before this if statement, that you are handling compressed (single byte) protocol field.

Also add the same comment before `net_buf_pull_u8()` as it is before `net_buf_pull_be16()` above:
```
		/* Remove the protocol field so that IP packet processing
		 * continues properly in net_core.c:process_data()
		 */
		(void)net_buf_pull_u8(pkt->buffer);
```